distributed-lock-core/src/main/java/com/github/alturkovic/lock/configuration/DistributedLockConfiguration.java

```diff
@@ -49,12 +50,12 @@
 
   @Bean
   @ConditionalOnMissingBean
-  public LockBeanPostProcessor lockBeanPostProcessor(final KeyGenerator keyGenerator,
-                                                     final ConfigurableBeanFactory configurableBeanFactory,
-                                                     final IntervalConverter intervalConverter,
-                                                     final RetriableLockFactory retriableLockFactory,
-                                                     @Autowired(required = false) final TaskScheduler taskScheduler) {
-    final LockBeanPostProcessor processor = new LockBeanPostProcessor(keyGenerator, configurableBeanFactory::getBean, intervalConverter, retriableLockFactory, taskScheduler);
+  public static LockBeanPostProcessor lockBeanPostProcessor(@Lazy final KeyGenerator keyGenerator,
```

Comment:
This is usually how it's done when declaring a bean of type `BeanPostProcessor` so that Spring will initialise it without eagerly initialising the outer `@Configuration` bean.
You can see an example of how Spring Boot registers `MeterRegistryPostProcessor`  in [MetricsAutoConfiguration](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsAutoConfiguration.java#L55).

Without `static`, Spring will log this statement:
> Bean 'com.github.alturkovic.lock.configuration.DistributedLockConfiguration' of type [com.github.alturkovic.lock.configuration.DistributedLockConfiguration$$EnhancerBySpringCGLIB$$bf5ccf21] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)

Reply:
Thanks for the explanation, appreciated!